Fix passmanager to preserve falsy pass outputs instead of reverting to input

neuralsorcerer opened the pull request:
Changes:
- Changed the fallback to only use the input IR when the pass returns `None`.

Why?
- The previous fallback used `ret or passmanager_ir`, which replaces any falsy return (e.g., 0, "", [], {}) with the original IR. Some passes legitimately produce an empty/zero IR (e.g., after pruning), so this silently reverted valid outputs.

jakelishman commented on qiskit/passmanager/base_tasks.py:
```diff
@@ -103,7 +103,7 @@ def execute(
             raise
         finally:
-            ret = ret or passmanager_ir
+            ret = passmanager_ir if ret is None else ret
```
Strictly this doesn't account for the possibility of `None` being a valid value within the backend's IR - we'd revert that as well if it were.  If you care about handling that case, we could make a sentinel `_UNSET = object()` global and then set `ret = _UNSET` and compare to `if ret is _UNSET` later.

neuralsorcerer replied:
updated

jakelishman reviewed (commented):
Thanks for this.  Was there an opened issue against Qiskit that corresponds to this PR? (No need to make one at this point if not, I'm just trying to make sure it gets closed if it exists.)

neuralsorcerer committed:
Fix passmanager to preserve falsy pass outputs instead of reverting to input

neuralsorcerer commented:
> Was there an opened issue against Qiskit that corresponds to this PR?

No opened issues against this. *(as per my knowledge)*

jakelishman committed:
Undo meaningless sentinel

I had originally suggested this, but it doesn't make sense; it would
need co-operation with the passes themselves.

jakelishman reviewed (approved):
Sorry, I don't know what I was thinking at all with that `_UNSET` thing - it doesn't make any sense in this situation.  It's a valid pattern for distinguishing certain returns from `None`, but in this case we'd have needed the tasks themselves to return the sentinel, which is a totally different thing (and not really necessary).

I reverted that change back to your original one, and this is good thanks!
